src/Simulation/Simulators/ResourcesEstimator/ResourcesEstimatorWithAdditionalPrimitiveOperations.cs

```diff
@@ -0,0 +1,92 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime;
+using Microsoft.Quantum.Simulation.Simulators;
+using Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators;
+
+namespace Simulator
+{
+    public class ResourcesEstimatorWithAdditionalPrimitiveOperations : ResourcesEstimator
```

Comment:
The logic in this class should be merged directly with the code in the [`ResourcesEstimator`](https://github.com/microsoft/qsharp-runtime/blob/main/src/Simulation/Simulators/ResourcesEstimator/ResourcesEstimator.cs) class.